src/include/txn_inline.h

```diff
@@ -264,8 +264,10 @@ __wt_txn_op_delete_apply_prepare_state(WT_SESSION_IMPL *session, WT_REF *ref, bo
     page_del = ref->page_del;
     if (page_del != NULL) {
         page_del->timestamp = ts;
-        if (commit)
+        if (commit) {
             page_del->durable_timestamp = txn->durable_timestamp;
+            page_del->committed = true;
```

Comment:
The fast truncate operation commit is a two-stage operation. In the first step, it processes any restored updates and later it marks the page_del structure with the committed flag. I feel instead of setting the `committed` flag here, move the entire logic of updating the `page_del` structure to the location where it marks the transaction as committed.

```
/*
     * Free updates.
     *
     * Resolve any fast-truncate transactions and allow eviction to proceed on instantiated pages.
     * This isn't done as part of the initial processing because until now the commit could still
     * switch to an abort. The action allowing eviction to proceed is clearing the WT_UPDATE list,
     * (if any), associated with the commit. We're the only consumer of that list and we no longer
     * need it, and eviction knows it means abort or commit has completed on instantiated pages.
     */
```

Reply:
I am thinking of getting rid of the committed flag in the page_del structure. It is kind of redundant information with prepared state and transaction id.
That way it will be more consistent with the way we handle update structure.
we can depend on prepare state to see whether page_del is committed or not instead of having this special flag.

Reply:
I don't remember the exact reason behind the split and the logic of committing the fast truncate operation in two stages. Better to check that logic before going on deleting the new flag.